frontend/packages/erd-core/src/components/ERDRenderer/TableDetailDrawer/TableDetailDrawer.tsx

```diff
@@ -4,34 +4,17 @@ import {
   useUserEditingStore,
 } from '@/stores'
 import { DrawerContent, DrawerPortal, DrawerRoot } from '@liam-hq/ui'
-import {
-  type FC,
-  type PropsWithChildren,
-  useCallback,
-  useEffect,
-  useState,
-} from 'react'
+import type { FC, PropsWithChildren } from 'react'
 // biome-ignore lint/nursery/useImportRestrictions: Fixed in the next PR.
 import { TableDetail } from '../ERDContent/TableNode/TableDetail'
 import styles from './TableDetailDrawer.module.css'
 
-const ANIMATION_DURATION = 165
+const handleClose = () => updateActiveTableName(undefined)
 
 export const TableDetailDrawerRoot: FC<PropsWithChildren> = ({ children }) => {
   const {
-    active: { tableName: tableId },
+    active: { tableName },
   } = useUserEditingStore()
-  const [open, setOpen] = useState(tableId !== undefined)
-
-  useEffect(() => {
-    setOpen(tableId !== undefined)
-  }, [tableId])
-
-  const handleClose = useCallback(() => {
-    setOpen(false)
-    // NOTE: Wait for the drawer to close before updating the active table name.
-    setTimeout(() => updateActiveTableName(undefined), ANIMATION_DURATION)
```

Comment:
This delayed function was executing after switching the active table by clicking on another TableNode, which was then closed.